### PR TITLE
FIX: Poll: real-time update of poll stats & clickable avatars with titles.

### PR DESCRIPTION
**Pull request by merefield**
* FIX: ensure voter count is updated in semi-real-time without browser refresh.  This was supported before but partially broke as a result of the Glimmer migration
* FIX: number polls results being incorrect when a remote voter supplied a new vote.
* FIX: restore ability to click an avatar and hover to get the title. (now broken out here: https://github.com/discourse/discourse/pull/28121)

This has required a refactor of the top layer of the new Glimmer implementation so we achieve a better "data-down" approach.

It appears to clean up the code nicely and provide a better user experience as remote updates to a poll in terms of results and voter count are reflected without a browser refresh being required.

@nattsw @jjaffeux @ZogStriP @davidtaylorhq

### CODE REVIEW

File: plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
 import { hbs } from "ember-cli-htmlbars";
```
Comment (jjaffeux) [inline plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js:1]: Sorry @merefield we are trying to move away from widgets, I can't accept a PR with 400 new lines of widget code. Maybe we try to extract smaller PRs out of this first please, I would really want to know why you think you need widgets here too.

Comment (davidtaylorhq) [reply plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js:1]: In theory there shouldn't be any need to use widgets at all. You can render a glimmer template directly using the `helper.renderGlimmer` function which is passed to `decorateCookedElement` callbacks.

e.g. https://github.com/discourse/discourse-policy/blob/3547813fe0cf25b53b7ef035f5ca6b207767862f/assets/javascripts/initializers/extend-for-policy.js#L50-L75

If something isn't working with that kind of no-widget strategy, then let's figure it out and fix it.

Comment (merefield) [reply plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js:1]: Avatar fix broken out here: #28121 28121

Comment (merefield) [reply plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js:1]: > In theory there shouldn't be any need to use widgets at all. You can render a glimmer template directly using the `helper.renderGlimmer` function which is passed to `decorateCookedElement` callbacks.

This is example what is already happening (prior to this PR).  There is nothing additional being rendered by widget code here, it's purely shifting the state management up to the widget level and changing the interface to the Glimmer Component so it is more granular and explicit.

**Comment by merefield** [comment]
I'll reraise with an alternative solution

**Pull request closed without merging**